src/parameterized_tendencies/gravity_wave_drag/non_orographic_gravity_wave.jl

```diff
@@ -157,56 +203,65 @@ function non_orographic_gravity_wave_tendency!(
 
     if model_config isa SingleColumnModel
         # source level: the index of the level that is closest to the source height
-        source_level = similar(Fields.level(Y.c.ρ, 1))
-        Fields.bycolumn(axes(ᶜρ)) do colidx
-            parent(source_level[colidx]) .=
-                argmin(abs.(parent(ᶜz[colidx]) .- gw_source_height))[1]
-        end
-        # damp level: for now we only deposit to top level for column setup
-        damp_level = similar(Fields.level(Y.c.ρ, 1))
-        Fields.bycolumn(axes(ᶜρ)) do colidx
-            parent(damp_level[colidx]) .= length(parent(ᶜz[colidx]))
+
+        Operators.column_mapreduce!(
+            reduce_fun1,
+            source_level_z,
+            ᶜz,
+            ᶜlevel,
+        ) do z, level
+            (abs.(z .- gw_source_height), level)
         end
+        source_level = source_level_z.:2
+
+        Operators.column_mapreduce!(sign, +, damp_level, ᶜz)
```

Comment:
@dennisYatunin  Would something like `@. damp_level = Spaces.nlevels(axes(ᶜz))` work?

Reply:
Yes, `damp_level .= Spaces.nlevels(axes(ᶜz))` or `fill!(damp_level, Spaces.nlevels(axes(ᶜz))` would be better here